File: sort/bubble.go
package sort

//冒泡排序
func BubbleSort(s []int) []int {
	length := len(s)
	for i := 0; i < length; i++ {
		for j := i + 1; j < length; j++ {
			if s[i] > s[j] {
				s[i], s[j] = s[j], s[i]
			}
		}
	}
	return s
}

File: py/structure/sort/bubble.py
# -*- coding:utf-8 -*-
"""
author:
date:
version:
content:
"""
from typing import List


# 冒泡排序
def bubble_sort(s: List) -> List:
    length = len(s)
    for i in range(length):
        for j in range(i + 1, length):
            if s[i] > s[j]:
                s[i], s[j] = s[j], s[i]
    return s
